### ISSUE TITLE
Should sampled store be derived?

### ISSUE THREAD

**Issue by AlexandrHoroshih** [effector/effector#605]
Lets say we have two derived stores - one via map and other via sample:
```ts
const $store = createStore(0)
const $map1 = $store.map(s => s + 1)
const $isItMapped = sample({
 source: $store,
 fn: s => s + 1
})
```

If we check their state in base graph, then their state will be, as expected, dervied from base store:
```ts
$map1.getState() // 1
$isItMapped.getState() // 1
```

But if we do this in forked scope with overriden value of base store:
```ts
const scope = fork({
 values: [[$store, 10]]
})

scope.getState($map1) // 11
scope.getState($isItMapped) // 1
```
state of mapped store will still have default value that is derived from base store, but state of "sampled" store will be the default one

So "sampled" store behaves like non-derived store, even though it is looks derived semantically 🤔 

Question: is it normal?
I would expect sampled store to be derived too, since it has explicit base stores in its definition via sample, but maybe i'm missing something 🤔

**Comment by AlexandrHoroshih** [comment]
Example:
https://share.effector.dev/FQBSzxgn

**Comment by sergeysova** [comment]
I think it is looks like a bug with sample

**Comment by igorkamyshev** [comment]
In my mind, `sampled`-store isn't *derived*. E.g., it should be serialized *if* we add `filter` or `clock` to its config.

**Comment by AlexandrHoroshih** [comment]
> In my mind, sampled-store isn't derived. E.g., it should be serialized if we add filter or clock to its config

Yes, in this case it is definetly not dervied, since its dependency rules is not non-conditional anymore

But in the simplest case it is non-conditional and semantically equal to mapped store (from my point of view)
```
const $isItMapped = sample({
 source: $store,
 fn: s => s + 1 // no other rules or conditions
})
```

**Comment by sergeysova** [comment]
If you try to add a conditions to sample, it will return `Event`.
So, we have a case where `sample({ source: $store, fn })` is an equivalent of `$store.map(fn)`
So, it is a derived store
